src/main/java/at/hannibal2/skyhanni/features/misc/IslandLeaveJoinMsgs.kt

```diff
@@ -0,0 +1,144 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.FriendAPI
+import at.hannibal2.skyhanni.data.GuildAPI
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.PartyAPI
+import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.events.TabListUpdateEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object IslandLeaveJoinMsgs {
+    private val config get() = SkyHanniMod.feature.misc.leaveJoinMsgs
+
+    private var players = mutableListOf<String>()
+    private var rawPlayersNew = mutableListOf<String>()
+    private var cleanPlayersNew = mutableListOf<String>()
+
+    private var updatedSinceWorldSwitch = false
+    private var onKnownIsland = false
+
+    private val rawPlayerPattern by RepoPattern.pattern(
+        "misc.islandleavejoinmsgs.rawplayers", "^§8\\[§\\w\\d+§8\\] (?<player>§\\w+).*$"
+    )
+    private val cleanPlayerPattern by RepoPattern.pattern(
+        "misc.islandleavejoinmsgs.cleanplayers", "^§8\\[§r§\\w\\d+§r§8\\] §r(?<player>§\\w+).*$"
+    )
+    private val offlinePlayerPattern by RepoPattern.pattern(
+        "misc.islandleavejoinmsgs.offlineplayers", "^§[0-9a-f]\\w+(?: §r§7\\(Offline [0-9dh+]+§r§7\\))?\$"
+    )
+    private val islandCategoryPattern by RepoPattern.pattern(
+        "misc.islandleavejoinmsgs.islandcategory", "^\\s+§r§b§lIsland$"
+    )
+    private val guestCategoryPattern by RepoPattern.pattern(
+        "misc.islandleavejoinmsgs.guestcategory", "^\\s+§r§5§lGuests §r§f\\(\\d+\\)$"
```

Comment:
The pattern key and the pattern should be on new lines also should all be part of a pattern group